server/portal/libs/agave/operations.py

```diff
@@ -442,18 +419,18 @@ def upload(client, system, path, uploaded_file):
     -------
     dict
     """
-
+    dest_path = os.path.join(path, uploaded_file.name)
     try:
-        listing(client, system=system, path=path)
+        file_listing = client.files.list(systemId=system, filePath=dest_path)
+        uploaded_file.name = increment_file_name(listing=file_listing, file_name=uploaded_file.name)
```

Comment:
Instead of reassigning a property of the function param, should we create a new variable?
```suggestion
        upload_name = increment_file_name(listing=file_listing, file_name=uploaded_file.name)
```